ast/format.go

```diff
@@ -376,9 +376,28 @@ func (f *formatter) formatIdentifier(n *Identifier) {
 }
 
 func (f *formatter) formatStringLiteral(n *StringLiteral) {
-	f.writeRune('"')
-	f.writeString(n.Value)
-	f.writeRune('"')
+	if n.Loc != nil && n.Loc.Source != "" {
+		// Preserve the exact literal if we have it
+		f.writeString(n.Loc.Source)
+	} else {
```

Comment:
This violates the Go style guide: https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow

Reply:
How? That guide is about errors, there is no error handling in the code. There are two mutually exclusive branches.

Reply:
You're right. I skimmed a bit too fast. At the same time, we generally try to return instead of using else in most other sections of the code as we apply that rule to more than just error handling.

Can you apply that same pattern here too even if there aren't errors?